onnx/defs/tensor/defs.cc

```diff
@@ -74,7 +74,7 @@ to equal sized parts.
 )DOC");
 
 OPERATOR_SCHEMA(Slice)
-    .NumInputs(1, 3)
+    .NumInputs(1)
```

Comment:
actually C2 implementation allows to specify starts and ends as arguments or inputs. Thus support for (3) here. @ezyang - do we rely on it anywhere?

Reply:
Why do we need both: attribute and input?

Reply:
The reason is exactly like adding `exponent` as input in Pow op: for dynamic use cases :-)

Reply:
It look weird to me to have one function support both instead of split them, and not all OPs require dynamic input. If that the case, we can promote all our attributes to input.s

Reply:
PyTorch's export which uses slice always uses slice's inputs as constants, and we currently fake up a tensor in order to support it. There are other operators that PyTorch supports which would make use of the dynamic behavior.

Reply:
Note that TF uses inputs for both `Transpose` and `Reshape` instead of attributes for the permutation / new shape.

Reply:
@ebarsoum Since there will be usecases that needs `starts` and `ends` to be dynamic, we need to add them back to Input (and remove them from attributes). Since "axes" is tied to "starts" and "ends', I believe it should be converted to input as well.